test/money/ecto/amount_type_test.exs

```diff
@@ -0,0 +1,64 @@
+defmodule Money.Ecto.Amount.TypeTest do
+  use ExUnit.Case, async: false
+
+  alias Money.Ecto.Amount.Type
+
+  doctest Type
+
+  setup do
+    Application.put_env(:money, :default_currency, :GBP)
+
+    on_exit fn ->
+      Application.delete_env(:money, :default_currency)
+    end
+  end
+
+  test "type/0" do
+    assert Type.type == :integer
+  end
+
+  test "cast/1 String" do
+    assert Type.cast("1000") == {:ok, Money.new(100000, :GBP)}
+    assert Type.cast("1234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("1,234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("£1234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("£1,234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("£ 1234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("£ 1,234.56") == {:ok, Money.new(123456, :GBP)}
+    assert Type.cast("£ 1234") == {:ok, Money.new(123400, :GBP)}
+    assert Type.cast("£ 1,234") == {:ok, Money.new(123400, :GBP)}
```

Comment:
Hi! I have a quick question about these tests and I guess how this would work in a real app. Say I have the following:

```
  1) test cast/1 String (Money.Ecto.Amount.TypeTest)
     test/money/ecto/amount_type_test.exs:20
     Assertion with == failed
     code:  assert Type.cast("R$ 1.234,00") == {:ok, Money.new(123400, :GBP)}
     left:  {:ok, %Money{currency: :GBP, amount: 123}}
     right: {:ok, %Money{currency: :GBP, amount: 123400}}
     stacktrace:
       test/money/ecto/amount_type_test.exs:30: (test)

```

Is this expected behavior? Thanks for creating the patch!

Reply:
This behavior depends on your config.
You can configure the separator and delimiter. 
e.g.:
```elixir
config :money,
  default_currency: :GDP,
  separator: ".",
  delimeter: ","

iex(2)> Type.cast("R$ 1.234,00")
{:ok, %Money{amount: 123400, currency: :USD}}
```

If you do not configure these parameters, you will get the result as in your example.
```elixir
config :money,
  default_currency: :GDP,

iex(2)> Type.cast("R$ 1.234,00")
{:ok, %Money{amount: 123, currency: :GBP}}
```

I will think about the default values of these configurations, but this will most require a change to the major version.

But the casting of other currencies in the default I think should not occur.

Reply:
Thanks for the quick reply!

> But the casting of other currencies in the default I think should not occur

That's fair! But, that's not what happens. It tries to cast, and the result is then incorrect, right? Should we warn/tell the user somehow that this happened? I can imagine a little "surprise" to know that amounts are flowing through. Maybe raise an error? I'm unsure, and regardless, thank you for explaining + doing this work!

Reply:
> I can imagine a little "surprise" to know that amounts are flowing through.

Agree. But I think it's subject for separate issue/PR. May you open such issue and describe all related edge cases which you know?

Reply:
I will fix cast in this PR.
Old Ecto type (deprecated `Money.Ecto.Type` ) will remain unchanged, and in this new Ecto type (`Money.Ecto.Amount.Type`) will be fixed.

Reply:
Completely agree with both of you. I don't know any edge cases other than the one I pointed out, almost by luck as I didn't know it was a bigger issue/beyond the scope of this patch. I'd love to help out in any way I can here, maybe an opportunity for me to contribute. But I think it would take longer to help/guide me than for you to tackle the problem. If you are ok with it taking longer than it would take you I'm happy to try! Are you on the Elixir Slack? If so, I'd love to ask more questions about this.

Here is the issue: https://github.com/liuggio/money/issues/98

Reply:
Yes, I am in Elixir Sluck (nick: nitrino). You can write to me there.

Reply:
Accept this PR in the current state and transfer the discussion to your issue.